packages/tx-construction/src/tx-builder/PolicyBuilder.ts

```diff
@@ -0,0 +1,63 @@
+import { AccountKeyDerivationPath, AsyncKeyAgent, KeyRole } from '@cardano-sdk/key-management';
+import { Cardano, nativeScriptPolicyId } from '@cardano-sdk/core';
+import { Ed25519KeyHashHex } from '@cardano-sdk/crypto';
+
+export class PolicyBuilder {
+  #keyAgent: AsyncKeyAgent;
+  #derivationPath: AccountKeyDerivationPath;
+  #derivationPathUpdated = false;
+  #policyScript: Cardano.NativeScript;
+  #keyHash: Ed25519KeyHashHex;
+
+  constructor(keyAgent: AsyncKeyAgent) {
+    this.#keyAgent = keyAgent;
+    this.#derivationPath = {
+      index: 0,
+      role: KeyRole.External
+    };
+  }
+
+  setDerivationPath(derivationPath: AccountKeyDerivationPath) {
+    this.#derivationPath = derivationPath;
+    this.#derivationPathUpdated = true;
+  }
+
+  getDerivationPath() {
+    return this.#derivationPath;
+  }
+
+  setPolicyScript(script: Cardano.NativeScript) {
+    this.#policyScript = script;
+    return this;
+  }
+
+  async getPolicyScript() {
+    if (!this.#policyScript) {
```

Comment:
If `derivationPath` is updated, the policy script should also be recalculated as it depends on the `keyHash`.
It would be great to have these cases covered in unit tests.
```
- if (!this.#policyScript) {
+ if (!this.#policyScript || this.#derivationPathUpdated) {

Reply:
updated in [c7e44c7](https://github.com/input-output-hk/cardano-js-sdk/pull/893/commits/c7e44c75007d6bbb5447b0889030b0005219a75c)